The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10846129566/job/30098460887) [(plain)](https://github.com/rust-lang/rust/commit/80b765b4367bf9fca5cda2dc65607b4f91c32b2d/checks/30098460887/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.906 Building wheels for collected packages: reuse
#16 2.907   Building wheel for reuse (pyproject.toml): started
#16 3.156   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 3.157   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 3.157   Stored in directory: /tmp/pip-ephem-wheel-cache-r0x2pux_/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 3.160 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.555 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.555 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.6s
---
    Checking regex-syntax v0.6.29
error[E0614]: type `DiagCtxt` cannot be dereferenced
   --> compiler/rustc_session/src/parse.rs:345:9
    |
345 |         *self.dcx = dcx;

    Checking miniz_oxide v0.7.4
   Compiling derive_more v0.99.18
    Checking regex v1.8.4
```

</details>

--- a/compiler/rustc_session/src/parse.rs
+++ b/compiler/rustc_session/src/parse.rs
@@ -340,4 +340,8 @@ impl ParseSess {
     pub fn dcx(&self) -> DiagCtxtHandle<'_> {
         self.dcx.handle()
     }
+
+    pub fn set_dcx(&mut self, dcx: DiagCtxt) {
+        self.dcx = dcx;
+    }
 }

